server/graphql/v2/mutation/IndividualMutations.ts

```diff
@@ -64,8 +69,11 @@ const individualMutations = {
       const account = await req.remoteUser.getCollective({ loaders: req.loaders });
       await TwoFactorAuthLib.enforceForAccount(req, account, { alwaysAskForToken: true });
 
-      // Check current password if one already set
-      if (req.remoteUser.passwordHash && req.jwtPayload?.scope !== 'reset-password') {
+      if (req.jwtPayload?.scope === 'reset-password') {
+        // If resetting with a reset token, the email must match
+        assert(req.jwtPayload['email'] === req.remoteUser.email, 'This token has expired');
```

Comment:
This will break any reset password links sent in the 75 minutes preceding the deploy. The alternative is to allow tokens without emails (`assert(!req.jwtPayload['email'] || req.jwtPayload['email'] === req.remoteUser.email`), but it did not seem worth the trouble.